### PR TITLE
New EVM node compatibility pipeline

### PR DESCRIPTION
**Pull request by Tofel**
## Motivation
@AnieeG made me do it

## Solution
I did it 🤷‍♂️

### CODE REVIEW

File: integration-tests/ccip-tests/testconfig/global.go [hunk 1]
```diff
@@ -151,13 +151,14 @@ func NewConfig() (*Config, error) {
...
-	NewCLCluster      *ChainlinkDeployment     `toml:",omitempty"` // NewCLCluster is the new chainlink cluster to create, if specified along with ExistingCLCluster this will be ignored
-	Network           *ctfconfig.NetworkConfig `toml:",omitempty"`
-	Logging           *ctfconfig.LoggingConfig `toml:"Logging"`
+	EnvUser                 string                               `toml:",omitempty"`
+	TTL                     *config.Duration                     `toml:",omitempty"`
+	ExistingCLCluster       *CLCluster                           `toml:",omitempty"` // ExistingCLCluster is the existing chainlink cluster to use, if specified it will be used instead of creating a new one
+	Mockserver              *string                              `toml:",omitempty"`
+	NewCLCluster            *ChainlinkDeployment                 `toml:",omitempty"` // NewCLCluster is the new chainlink cluster to create, if specified along with ExistingCLCluster this will be ignored
+	Network                 *ctfconfig.NetworkConfig             `toml:",omitempty"`
+	PrivateEthereumNetworks map[string]*test_env.EthereumNetwork `toml:",omitempty"`
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testconfig/global.go:160]: can it not be handled with `Network`?

File: integration-tests/ccip-tests/testsetups/test_env.go [hunk 2]
```diff
@@ -173,25 +173,50 @@ func DeployLocalCluster(
...
 ) (*test_env.CLClusterTestEnv, func() error) {
 	selectedNetworks := testInputs.SelectedNetworks
+
+	// get TOML config, but if it's not present use selected networks to generate a default one (2 geth pow chains on 1.12)
+	privateEthereumNetworks := []*ctftestenv.EthereumNetwork{}
+
+	require.True(t, len(testInputs.EnvInput.PrivateEthereumNetworks) == 0 || len(testInputs.EnvInput.PrivateEthereumNetworks) == len(testInputs.SelectedNetworks), "You either have to configure all private networks or none at all")
+
+	// if there's no configuration use default
+	if len(testInputs.EnvInput.PrivateEthereumNetworks) == 0 {
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testsetups/test_env.go]: To make it consistent with other defaults let's use ccip-default.toml to set the defaults. All the other defaults are set with ccip-default.toml.

Comment (AnieeG) [reply integration-tests/ccip-tests/testsetups/test_env.go]: Or you can get rid of `PrivateEthereumNetworks` and always handle the input from `selected_networks`.

File: integration-tests/ccip-tests/testconfig/global.go (hunk 1 above)
Comment (Tofel) [reply integration-tests/ccip-tests/testconfig/global.go:160]: it cannot,  it has nothing to do with `Network`. The only thing they have in common in chain id.

**Review by AnieeG — APPROVED** [review]

File: integration-tests/ccip-tests/testsetups/test_env.go [hunk 3]
```diff
@@ -173,25 +173,50 @@ func DeployLocalCluster(
...
-	for _, n := range env.PrivateChain {
-		primaryNode := n.GetPrimaryNode()
-		require.NotNil(t, primaryNode, "Primary node is nil in PrivateChain interface")
-		for i, networkCfg := range selectedNetworks {
-			if networkCfg.ChainID == n.GetNetworkConfig().ChainID {
-				selectedNetworks[i].URLs = []string{primaryNode.GetInternalWsUrl()}
-				selectedNetworks[i].HTTPURLs = []string{primaryNode.GetInternalHttpUrl()}
-			}
-		}
-		testInputs.SelectedNetworks = selectedNetworks
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testsetups/test_env.go:193]: This line might be still needed.

File: integration-tests/ccip-tests/testsetups/ccip.go [hunk 4]
```diff
@@ -184,45 +184,43 @@ func (c *CCIPTestConfig) SetNetworkPairs(lggr zerolog.Logger) error {
...
-		if simulated {
-			actualNoOfNetworks := len(c.SelectedNetworks)
-			n := c.SelectedNetworks[0]
-			var chainIDs []int64
-			for _, id := range chainselectors.TestChainIds() {
-				if id == 2337 {
-					continue
-				}
-				chainIDs = append(chainIDs, int64(id))
+	if simulated && len(c.SelectedNetworks) < c.TestGroupInput.NoOfNetworks {
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testsetups/ccip.go:187]: 👍

**Review by AnieeG — APPROVED** [review]
LGTM

**Pull request merged**